release-prep.sh

```diff
@@ -0,0 +1,211 @@
+		fi
+		if [[ "$2" = *"m"* ]]
+		then
+            genManpages=false
+		fi
+		shift
+	    else
+		echo 'Error: "--skip" requires an argument'
+		exit 1
+	    fi
+	    ;;
+	*)               # Default case: If no more options then break out of the loop.
+        break
+    esac
+    shift
+done
+
+if [[ $verBump = true ]]
+then
+    # Bump Version numbers
+    # Get version
+    if [[ -n "$1" ]]
+    then
+        VERSION=$1
+        shift
+    fi
+
+    # Check that a version is specified
+    if [[ $VERSION == "" ]]
+    then
+        echo "$scriptName: Missing version."
+        echo "Try $scriptName --help for more information"
+        exit 1
+    fi
+
+    echo "Setting Version number"
+    major=$(echo $VERSION | cut -d. -f1)
+    minor=$(echo $VERSION | cut -d. -f2)
+    rev=$(echo $VERSION | cut -d. -f3)
+
+    # configure.ac
+    sed -i "/define(_CLIENT_VERSION_MAJOR, /c\define(_CLIENT_VERSION_MAJOR, $major)" ./configure.ac
+    sed -i "/define(_CLIENT_VERSION_MINOR, /c\define(_CLIENT_VERSION_MINOR, $minor)" ./configure.ac
+    sed -i "/define(_CLIENT_VERSION_REVISION, /c\define(_CLIENT_VERSION_REVISION, $rev)" ./configure.ac
+    sed -i "/define(_CLIENT_VERSION_IS_RELEASE, /c\define(_CLIENT_VERSION_IS_RELEASE, true)" ./configure.ac
+
+    # src/clientversion.h
+    sed -i "/#define CLIENT_VERSION_MAJOR /c\#define CLIENT_VERSION_MAJOR $major" ./src/clientversion.h
+    sed -i "/#define CLIENT_VERSION_MINOR /c\#define CLIENT_VERSION_MINOR $minor" ./src/clientversion.h
+    sed -i "/#define CLIENT_VERSION_REVISION /c\#define CLIENT_VERSION_REVISION $rev" ./src/clientversion.h
+    sed -i "/#define CLIENT_VERSION_IS_RELEASE /c\#define CLIENT_VERSION_IS_RELEASE true" ./src/clientversion.h
+
+    # docs
+    sed -i "/PROJECT_NUMBER         = /c\PROJECT_NUMBER         = $VERSION" ./doc/Doxyfile
+    sed -i "1s/.*/Bitcoin Core $VERSION/" ./doc/README.md
+    sed -i "1s/.*/Bitcoin Core $VERSION/" ./doc/README_windows.txt
+
+    # gitian descriptors
+    sed -i "2s/.*/name: \"bitcoin-win-$major.$minor\"/" ./contrib/gitian-descriptors/gitian-win.yml
+    sed -i "2s/.*/name: \"bitcoin-linux-$major.$minor\"/" ./contrib/gitian-descriptors/gitian-linux.yml
+    sed -i "2s/.*/name: \"bitcoin-osx-$major.$minor\"/" ./contrib/gitian-descriptors/gitian-osx.yml
+fi
+
+if [[ $chainparamsUpdate = true ]]
+then
+    # Update nMinimumChainWork and defaultAssumeValid
+    echo "Updating nMinimumChainWork and defaultAssumeValid"
+    blockchaininfo=`src/bitcoin-cli ${DATADIR} getblockchaininfo`
+    chainwork=`echo "$blockchaininfo" | jq -r '.chainwork'`
+    bestblockhash=`echo "$blockchaininfo" | jq -r '.bestblockhash'`
+    sed -i "0,/        consensus.nMinimumChainWork = uint256S(.*/s//        consensus.nMinimumChainWork = uint256S(\"0x$chainwork\");/" ./src/chainparams.cpp
+    sed -i "0,/        consensus.defaultAssumeValid = uint256S(.*/s//        consensus.defaultAssumeValid = uint256S(\"0x$bestblockhash\");/" ./src/chainparams.cpp
```

Comment:
I think this is a manual process for each network and should probably not be automated.